### PR TITLE
fix#23722 Signature mismatch of get_formatted_order_total

### LINKED ISSUE #23722: Signature mismatch of get_formatted_order_total
**Describe the bug**
Method call uses the 1 parameter,  but the method signature uses 0 parameters for the method get_formatted_order_total. 

**To Reproduce**
1. Go to 'plugins\woocommerce\includes\abstracts\abstract-wc-order.php'
2. Find the method call get_formatted_order_total and method signature.

**Screenshots**
Screenshot_1
Screenshot_2

**Expected behavior**
A parameter should not pass to the function Or signature have the parameter.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by ajayghaghretiya**
### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #23722 .

### How to test the changes in this Pull Request:

1. Go to 'plugins\woocommerce\includes\abstracts\abstract-wc-order.php'
2. Find the method call get_formatted_order_total and method signature.

> updated the get_formatted_order_total method signature.

### CODE REVIEW

File: includes/abstracts/abstract-wc-order.php [hunk 1]
```diff
@@ -1668,9 +1668,11 @@ public function get_formatted_line_subtotal( $item, $tax_display = '' ) {
 	/**
 	 * Gets order total - formatted for display.
 	 *
+	 * @param string $tax_display Excl or incl tax display mode.
+	 *
 	 * @return string
 	 */
-	public function get_formatted_order_total() {
+	public function get_formatted_order_total( $tax_display ) {
```
Comment (claudiosanches) [inline includes/abstracts/abstract-wc-order.php]: It's missing default value and `$display_refunded`,

This is how suppose to be:

```php
	/**
	 * Gets order total - formatted for display.
	 *
	 * @param string $tax_display      Type of tax display.
	 * @param bool   $display_refunded If should include refunded value.
	 *
	 * @return string
	 */
	public function get_formatted_order_total( $tax_display = '', $display_refunded = true ) {
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Thanks for your help. Just need to do a small fix and this PR will be ready to go.

**Comment by Mr-AjayM** [comment]
Hello @claudiosanches 

Thanks for sharing your inputs for this!

I have updated the changes, kindly review it and let me know if we need to do any additional changes for this.

Thanks & Regards,
Aj

**Comment by kloon** [comment]
Curious if this was throwing an error, and if so how to replicate that? The parameters are not used anywhere in the function in WC_Abstract_Order so seems useless to include them?

The function is already extended in WC_Order which extends WC_Abstract_Order so when it is called in WC_Abstract_Order it is actually calling the function in WC_Order.

**Comment by Mr-AjayM** [comment]
@kloon It's showing me this warning when running the PHPCS with the WooCommerce Core standards. I am not sure either solve by removing the parameter when calling the function of adding the parameters in the definition of the function.  

That also makes sense that we are not using any parameter in the method as well. Let's wait for other feedback as well.

**Comment by claudiosanches** [comment]
@kloon depends on the PHP version and configuration, there's changes on it on PHP 7 too, see: https://stackoverflow.com/questions/36079651/silence-declaration-should-be-compatible-warnings-in-php-7
Also we have some fixes like that in the past: https://github.com/woocommerce/woocommerce/pull/3420

**Comment by claudiosanches** [comment]
@Mr-AjayM what PHP version are you using on your environment?

**Comment by Mr-AjayM** [comment]
@claudiosanches I am on PHP version 7.0.10 in my local environment.

**Comment by claudiosanches** [comment]
Testing it better, in fact this change may cause a PHP warning, if extending this class:

```
PHP Warning:  Declaration of WC_Order::get_formatted_order_total() should be compatible with WC_Abstract_Order::get_formatted_order_total($tax_display = '', $display_refunded = true) in wp-content/plugins/woocommerce/includes/class-wc-order.php on line 0
```

We can't reproduce this the reported error, so let's leave close for now, if we get more reports like that, we'll update the abstract class.
Thanks for your help.

**Pull request closed without merging**